### DISCUSSION TITLE
Missing S3 configuration options

### DISCUSSION THREAD

**Discussion by hmmhmmhm** [keystonejs/keystone#8907] [Feature requests]
https://keystonejs.com/docs/config/config#storage-images-and-files

`signatureVersion` cannot be set in the file storage of Keystone 6.

Cloudflare R2 Storage requires setting 'signatureVersion: 'v4'.
It should be able to add more diverse S3 options.

**Comment by hmmhmmhm** [comment]
I have confirmed that Cloudflare R2 is binding normally without the `signatureVersion` setting.

**Comment by pc-erin** [comment]
Adding to this, the ability to set parameters like ContentType and ContentDisposition would be extremely helpful.

I think we could get all of this implemented at once by adding an `s3Params` option to `kind: 's3'` file and image configs.

If it's an object, we just merge it into the S3 Upload params, and if it's a function we pass in things like the file name and size, then merge the returned object into the upload params.

I could add this but I'm not familiar with how to test added features on keystone.
